deploy/MigrationInit.sol

```diff
@@ -0,0 +1,157 @@
+        AuthedLike(dss.chainlog.getAddress("MCD_PAUSE")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("SPLITTER_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("OSM_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("CLIPPER_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("DIRECT_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("STARKNET_ESCROW_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("LINE_MOM")).setAuthority(inst.chief);
+        AuthedLike(dss.chainlog.getAddress("LITE_PSM_MOM")).setAuthority(inst.chief);
+        dss.chainlog.setAddress("MCD_ADM", inst.chief);
+
+        // New VoteDelegate factory (must be done before initLockstake)
+        dss.chainlog.setAddress("VOTE_DELEGATE_FACTORY_LEGACY", dss.chainlog.getAddress("VOTE_DELEGATE_FACTORY"));
+        dss.chainlog.setAddress("VOTE_DELEGATE_FACTORY", inst.voteDelegateFactory);
+
+        // New MKR to SKY migrator (must be done before initLockstake)
+        SkyInit.updateMkrSky(dss, inst.mkrSky);
+
+        // Set the new SKY medianizer in the flapper
+        // Note: we assume the current flapper is given permission to read from the medianizer
+        FlapperLike(dss.chainlog.getAddress("MCD_FLAP")).file("pip", cfg.skyMedianizer);
+        dss.chainlog.setAddress("FLAP_SKY_ORACLE", cfg.skyMedianizer);
+
+        // New SKY OSM (must be done before initLockstake)
+        // Note: we assume the OSM is given permission to read from the medianizer
+        // Note: the rest of the OSM setup is done in initLockstake below
+        // Note: PIP_MKR is still used in the old lockstake, so we don't remove it yet from the chainlog
+        dss.chainlog.setAddress("PIP_SKY", inst.skyOsm);
+
+        // Init new farm to be used in lockstake
+        // Note: we assume that if splitter.burn, splitter.hop, vow.bump or vow.hump are modified it is done outside this lib
+        // Note: we assume cfg.rewardsDuration to be same as the final value of splitter.hop for the spell
+        address splitter = dss.chainlog.getAddress("MCD_SPLIT");
+        StakingRewardsInit.init(inst.lsskyUsdsFarm, StakingRewardsInitParams({dist: splitter}));
+        StakingRewardsLike(inst.lsskyUsdsFarm).setRewardsDuration(cfg.rewardsDuration);
+        SplitterLike(splitter).file("farm", inst.lsskyUsdsFarm);
+        dss.chainlog.setAddress("REWARDS_LSSKY_USDS", inst.lsskyUsdsFarm);
+        dss.chainlog.setAddress("REWARDS_LSMKR_USDS_LEGACY", dss.chainlog.getAddress("REWARDS_LSMKR_USDS"));
+        dss.chainlog.removeAddress("REWARDS_LSMKR_USDS");
+
+        // Lockstake Migration
+        LockstakeInit.initLockstake(dss, inst.lockstakeInstance, cfg.lockstakeConfig);
+
+        // MKR flops are turned off
+        VowLike(dss.chainlog.getAddress("MCD_VOW")).file("sump", type(uint256).max);
```

Comment:
Could also de-auth the Vow on the Flopper. Although this is sufficient for all practical purposes.